common/models/user.js

```diff
@@ -382,14 +371,105 @@ module.exports = function(User) {
     }
   );
 
+  User.requestAuthLink = function requestAuthLink(email) {
+    if (!isEmail(email)) {
+      return Promise.reject(
+        new Error('The submitted email not valid.')
+      );
+    }
+
+    var userObj = {
+      username: 'fcc' + uuid.v4().slice(0, 8),
+      email: email,
+      emailVerified: false
+    };
+    return User.findOrCreate$({ where: { email }}, userObj)
+      .map(([ err, user, isCreated ]) => {
```

Comment:
Errors here will be passed down the pipeline until a catch operator is hit or a the onError handler in the subscribe method is called. The err is not passed into the map operator.